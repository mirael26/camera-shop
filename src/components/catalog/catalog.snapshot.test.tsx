import axios from 'axios';
import Catalog from './catalog';
import { renderWithReduxAndRouter } from '../../test/helpers/render-with-redux-and-router';
import { productsMock } from '../../test/mocks';

jest.mock('./filters/filters', () => 'Filters');
jest.mock('./pagination/pagination', () => 'Pagination');
jest.mock('./sorts/sorts', () => 'Sorts');
jest.mock('../product-card/product-card', () => 'ProductCard');
jest.mock('axios');

test('Catalog renders correctly', () => {
  jest.mocked(axios).get.mockResolvedValue({ data: productsMock });
  const { asFragment } = renderWithReduxAndRouter(<Catalog/>, {initialState: { data: { products: productsMock }}});
  expect(asFragment()).toMatchSnapshot();
});
