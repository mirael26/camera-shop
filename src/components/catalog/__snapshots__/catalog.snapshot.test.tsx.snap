// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog renders correctly 1`] = `
<DocumentFragment>
  <section
    class="catalog"
  >
    <div
      class="container"
    >
      <h1
        class="title title--h2"
      >
        Каталог фото- и видеотехники
      </h1>
      <div
        class="page-content__columns"
      >
        <div
          class="catalog__aside"
        >
          <filters />
        </div>
        <div
          class="catalog__content"
        >
          <sorts />
          <productlist
            productscountonpage="9"
          />
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
